schemas/pulse-job.yml

```diff
@@ -204,28 +241,180 @@ properties:
   runMachine:
     $ref: "#/definitions/machine"
 
-  artifacts:
-    type: "array"
+  textLogSummary:
+    type: object
+    description: |
+      This object defines what is seen in the Treeherder Log Viewer.  This object
+      can be submitted here, or it will be generated from the submitted log.
+    properties:
+      step_data:
+        type: object
+        properties:
+          all_errors:
+            description: |
+              Lists all the lines of the log that have errors.  These line
+              values will be used the the process that generates the
+              ``Bug suggestions`` for display on the job summary panel in
+              Treeherder.
+            type: array
+            items:
+              type: object
+              properties:
+                line:
+                  type: string
+                  minLength: 1
+                  maxLength: 255
+                linenumber:
+                  type: integer
+              additionalProperties: false
+              required:
+              - line
+              - linenumber
+            additionalItems: false
+          steps:
+            type: array
+            items:
+              type: object
+              properties:
+                errors:
+                  type: array
+                  items:
+                    type: object
+                    properties:
+                      line:
+                        type: string
+                        minLength: 1
+                        maxLength: 255
+                      linenumber:
+                        type: integer
+                        minimum: 0
+                    additionalProperties: false
+                  additionalItems: false
+                name:
+                  type: string
+                  minLength: 1
+                  maxLength: 255
+                started:
+                  type: string
+                  format: date-time
+                started_linenumber:
+                  type: integer
+                  minimum: 0
+                finished_linenumber:
+                  type: integer
+                  minimum: 0
+                finished:
+                  type: string
+                  format: date-time
+                error_count:
+                  type: integer
+                  minimum: 0
+                duration:
+                  type: integer
+                  minimum: 0
+                order:
+                  type: integer
+                  minimum: 0
```

Comment:
What is the `order`? if this is the order in which the step comes, I strongly advice you remove it and require people to define the `steps` array as an ordered array.